lib/improver/metadata/utilities.py

```diff
@@ -44,16 +42,14 @@ def generate_hash(data_in):
     Args:
         data_in (any):
             The data from which a hash is to be generated. This can be of any
-            type that can be pickled.
+            type that can be pretty printed.
     Returns:
-        hash (str):
-            A hexidecimal hash representing the data.
+        hash (string):
+            A hexadecimal string which is a hash hexdigest of the data as a
+            string.
     """
-    hashable_type = pickle.dumps(data_in)
-    # Marked as 'nosec' as the usage of MD5 hash is to produce a good checksum,
-    # rather than for cryptographic hashing purposes
-    hash_result = hashlib.md5(hashable_type).hexdigest()  # nosec
-    return hash_result
+    bytestring = pprint.pformat(data_in, sort_dicts=True).encode('utf-8')
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unexpected keyword argument 'sort_dicts' in function call](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4416639)